### ISSUE TITLE
basemap dependency causes Read the Docs builds to fail

### ISSUE THREAD

**Issue by weaverba137** [desihub/desiutil#53]
Now that basemap is a dependency for desiutil, it is causing [documentation builds](https://desiutil.readthedocs.io) to fail.  Here's the traceback:

```
Collecting git+https://github.com/matplotlib/basemap.git (from -r /home/docs/checkouts/readthedocs.org/user_builds/desiutil/checkouts/latest/requirements.txt (line 7))
  Cloning https://github.com/matplotlib/basemap.git to /tmp/pip-rp95hnvj-build
    Complete output from command python setup.py egg_info:
    checking for GEOS lib in /home/docs ....
    checking for GEOS lib in /home/docs/local ....
    checking for GEOS lib in /usr ....
    checking for GEOS lib in /usr/local ....
    checking for GEOS lib in /sw ....
    checking for GEOS lib in /opt ....
    checking for GEOS lib in /opt/local ....
    
    Can't find geos library in standard locations ('/home/docs', '/home/docs/local', '/usr', '/usr/local', '/sw', '/opt', '/opt/local').
    Please install the corresponding packages using your
    systems software management system (e.g. for Debian Linux do:
    'apt-get install libgeos-3.3.3 libgeos-c1 libgeos-dev' and/or
    set the environment variable GEOS_DIR to point to the location
    where geos is installed (for example, if geos_c.h
    is in /usr/local/include, and libgeos_c is in /usr/local/lib,
    set GEOS_DIR to /usr/local), or edit the setup.py script
    manually and set the variable GEOS_dir (right after the line
    that says "set GEOS_dir manually here".
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-rp95hnvj-build/
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Command time: 46s Return: 1
```

**Comment by sbailey** [comment]
Since many of our packages depend upon desiutil, ot would be highly preferably if healpix and basemap could be treated as optional dependencies: if you are using the sky plotting tools you need them, but otherwise desiutil should be completely functional without them, including building documentation.

**Comment by tskisner** [comment]
One note:  healpy has build problems on some systems (like NERSC), since it builds the C++ healpix library internally but that code needs additional compile options which cannot be specified.  I actually got tired of the healpix build system so much that I repackaged it with autotools:  https://github.com/tskisner/healpix-autotools

That is the version that I use at NERSC, so I guess I could include healpix/healpy in the list of "extra" packages that are installed with desi-conda.  The healpix developers know of my work, but it is more of a "loose collective" rather than a coordinated development effort between all the language flavors of healpix.
